pinot-core/src/test/java/org/apache/pinot/core/data/manager/realtime/LLRealtimeSegmentDataManagerTest.java

```diff
@@ -128,9 +131,9 @@ private TableConfig createTableConfig()
     return JsonUtils.stringToObject(_tableConfigJson, TableConfig.class);
   }
 
-  private RealtimeTableDataManager createTableDataManager() {
+  private RealtimeTableDataManager createTableDataManager(TableConfig tableConfig) {
```

Comment:
Can you add the test for the particular issue that we had if possible?

1. Create 2 data managers for 2 tables.
2. Call the shutdown() from one data manager.
3. Check if the lease extender for 2nd data manager is still alive. (Also double check if this test would fail before the fix)

Reply:
Updated the test to check the executor is not removed when tableDataManager is removed. Checked manually that the executor is removed before the fix.